internal/types/compare_order.go

```diff
@@ -91,9 +92,35 @@ const (
 	Descending SortType = -1
 )
 
+// numberOrderResult represents the comparison order of numbers.
+type numberOrderResult uint8
+
+const (
+	_ numberOrderResult = iota
+	doubleDT
+	int32DT
+	int64DT
+)
+
+// detectNumberType returns a sequence for float64, int32 and int64 types.
+func detectNumberType(value any) numberOrderResult {
```

Comment:
I see where this function is used, but I don't understand how it works. The result of this function is compared with `numbersDataType` which is another constant, and there are no other use cases for it.

Should this function be called something like `isNumber` then and include a comparison with `numbersDataType`?